app/decorators/models/product/add_relation_methods.rb

```diff
@@ -8,7 +8,7 @@ def self.prepended(base)
 
         # When a Spree::Product is destroyed, we also want to destroy all Spree::Relations
         # "from" it as well as "to" it.
-        base.after_destroy :destroy_product_relations
+        base.after_discard :destroy_product_relations
```

Comment:
I think that this works only on versions of solidus that have the `discard` gem already. Would you mind adding a check that does that conditionally if `after_discard` is defined?

Reply:
Just updated the branch. Will that suffice?

Reply:
Yes I think this works, thanks!